.github/workflows/container-image.yml

```diff
@@ -0,0 +1,34 @@
+name: Build and Push Container Image
+
+on:
+  push:
+    branches:
+      - main
+      - buildanduploadimage
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+      
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+    
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    
+    - name: Build and push
+      uses: docker/build-push-action@v2
+      with:
+        context: .
+        push: true
+        tags: ghcr.io/${{ github.repository_owner }}/arxivchatguru:${{ github.ref == 'refs/heads/main' && 'latest' || 'canary' }}
```

Comment:
can you share more about the versioning logic?

Reply:
We can version as you want. Tbh, I did not pay attention to it further the fact that I wanted to test it and not in the main branch.

Reply:
I think this works for now, thanks!